### ISSUE TITLE
angular 6 service not found

### ISSUE THREAD

**Issue by keslavi** [kuuurt13/ng-block-ui#71]
using angular-cli, v6, and trying to use ng-block-ui.
getting service not found error message... possibly a difference between v5 and 6? 

(added rxjs-compat btw)

**Comment by keslavi** [comment]
Thank you for your help :) unsure if this is an ng6/ng5 thing or not. 

**console:** 
ERROR in node_modules/@angular/core/src/render3/ng_dev_mode.d.ts(9,11): error TS2451: Cannot redeclare block-scoped variable 'ngDevMode'.
node_modules/ng-block-ui/node_modules/@angular/core/src/render3/ng_dev_mode.d.ts(9,11): error TS2451: Cannot redeclare block-scoped variable 'ngDevMode'.

**In ng_dev_mode.d.ts, i commented out // const ngDevMode: boolean; to continue.** 

i ?wdm?: Failed to compile.
i ?wdm?: Compiling...

Date: 2018-07-02T13:55:55.594Z - Hash: 949cd8775bde31eacf89 - Time: 19730ms
3 unchanged chunks
chunk {main} main.js, main.js.map (main) 707 kB [initial] [rendered]
chunk {polyfills} polyfills.js, polyfills.js.map (polyfills) 473 kB [initial] [rendered]
chunk {vendor} vendor.js, vendor.js.map (vendor) 7.41 MB [initial] [rendered]

WARNING in ./node_modules/ng-block-ui/node_modules/@angular/core/esm5/core.js
6570:15-36 Critical dependency: the request of a dependency is an expression

WARNING in ./node_modules/ng-block-ui/node_modules/@angular/core/esm5/core.js
6590:15-102 Critical dependency: the request of a dependency is an expression
i ?wdm?: Compiled with warnings.

**then, in the browser, i see this:** 

ERROR Error: StaticInjectorError(AppModule)[AppComponent -> BlockUIService]: 
  StaticInjectorError(Platform: core)[AppComponent -> BlockUIService]: 
    NullInjectorError: No provider for BlockUIService!
    at NullInjector.push../node_modules/@angular/core/fesm5/core.js.NullInjector.get (core.js:936)
    at resolveToken (core.js:1175)
    at tryResolveToken (core.js:1120)
    at StaticInjector.push../node_modules/@angular/core/fesm5/core.js.StaticInjector.get (core.js:1015)
    at resolveToken (core.js:1175)
    at tryResolveToken (core.js:1120)
    at StaticInjector.push../node_modules/@angular/core/fesm5/core.js.StaticInjector.get (core.js:1015)
    at resolveNgModuleDep (core.js:8065)
    at NgModuleRef_.push../node_modules/@angular/core/fesm5/core.js.NgModuleRef_.get (core.js:8753)
    at resolveDep (core.js:9118)
View_AppComponent_Host_0 @ AppComponent_Host.ngfactory.js? [sm]:1
proxyClass @ compiler.js:10058
push../node_modules/@angular/core/fesm5/core.js.DebugContext_.logError @ core.js:11002
push../node_modules/@angular/core/fesm5/core.js.ErrorHandler.handleError @ core.js:1567
(anonymous) @ core.js:4253
push../node_modules/zone.js/dist/zone.js.ZoneDelegate.invoke @ zone.js:388
push../node_modules/zone.js/dist/zone.js.Zone.run @ zone.js:138
push../node_modules/@angular/core/fesm5/core.js.NgZone.runOutsideAngular @ core.js:3613
(anonymous) @ core.js:4253
push../node_modules/zone.js/dist/zone.js.ZoneDelegate.invoke @ zone.js:388
onInvoke @ core.js:3654
push../node_modules/zone.js/dist/zone.js.ZoneDelegate.invoke @ zone.js:387
push../node_modules/zone.js/dist/zone.js.Zone.run @ zone.js:138
(anonymous) @ zone.js:872
push../node_modules/zone.js/dist/zone.js.ZoneDelegate.invokeTask @ zone.js:421
onInvokeTask @ core.js:3645
push../node_modules/zone.js/dist/zone.js.ZoneDelegate.invokeTask @ zone.js:420
push../node_modules/zone.js/dist/zone.js.Zone.runTask @ zone.js:188
drainMicroTaskQueue @ zone.js:595
Promise.then (async)
scheduleMicroTask @ zone.js:578
push../node_modules/zone.js/dist/zone.js.ZoneDelegate.scheduleTask @ zone.js:410
push../node_modules/zone.js/dist/zone.js.Zone.scheduleTask @ zone.js:232
push../node_modules/zone.js/dist/zone.js.Zone.scheduleMicroTask @ zone.js:252
scheduleResolveOrReject @ zone.js:862
ZoneAwarePromise.then @ zone.js:962
push../node_modules/@angular/core/fesm5/core.js.PlatformRef.bootstrapModule @ core.js:4183
./src/main.ts @ main.ts:12
__webpack_require__ @ bootstrap:76
0 @ polyfills.ts:79
__webpack_require__ @ bootstrap:76
checkDeferredModules @ bootstrap:43
webpackJsonpCallback @ bootstrap:30
(anonymous) @ main.js:1
AppComponent_Host.ngfactory.js? [sm]:1 ERROR CONTEXT DebugContext_ {view: {…}, nodeIndex: 1, nodeDef: {…}, elDef: {…}, elView: {…}}
View_AppComponent_Host_0 @ AppComponent_Host.ngfactory.js? [sm]:1
proxyClass @ compiler.js:10058
push../node_modules/@angular/core/fesm5/core.js.DebugContext_.logError @ core.js:11002
push../node_modules/@angular/core/fesm5/core.js.ErrorHandler.handleError @ core.js:1572
(anonymous) @ core.js:4253
push../node_modules/zone.js/dist/zone.js.ZoneDelegate.invoke @ zone.js:388
push../node_modules/zone.js/dist/zone.js.Zone.run @ zone.js:138
push../node_modules/@angular/core/fesm5/core.js.NgZone.runOutsideAngular @ core.js:3613
(anonymous) @ core.js:4253
push../node_modules/zone.js/dist/zone.js.ZoneDelegate.invoke @ zone.js:388
onInvoke @ core.js:3654
push../node_modules/zone.js/dist/zone.js.ZoneDelegate.invoke @ zone.js:387
push../node_modules/zone.js/dist/zone.js.Zone.run @ zone.js:138
(anonymous) @ zone.js:872
push../node_modules/zone.js/dist/zone.js.ZoneDelegate.invokeTask @ zone.js:421
onInvokeTask @ core.js:3645
push../node_modules/zone.js/dist/zone.js.ZoneDelegate.invokeTask @ zone.js:420
push../node_modules/zone.js/dist/zone.js.Zone.runTask @ zone.js:188
drainMicroTaskQueue @ zone.js:595
Promise.then (async)
scheduleMicroTask @ zone.js:578
push../node_modules/zone.js/dist/zone.js.ZoneDelegate.scheduleTask @ zone.js:410
push../node_modules/zone.js/dist/zone.js.Zone.scheduleTask @ zone.js:232
push../node_modules/zone.js/dist/zone.js.Zone.scheduleMicroTask @ zone.js:252
scheduleResolveOrReject @ zone.js:862
ZoneAwarePromise.then @ zone.js:962
push../node_modules/@angular/core/fesm5/core.js.PlatformRef.bootstrapModule @ core.js:4183
./src/main.ts @ main.ts:12
__webpack_require__ @ bootstrap:76
0 @ polyfills.ts:79
__webpack_require__ @ bootstrap:76
checkDeferredModules @ bootstrap:43
webpackJsonpCallback @ bootstrap:30
(anonymous) @ main.js:1
main.ts:13 Error: StaticInjectorError(AppModule)[AppComponent -> BlockUIService]: 
  StaticInjectorError(Platform: core)[AppComponent -> BlockUIService]: 
    NullInjectorError: No provider for BlockUIService!
    at NullInjector.push../node_modules/@angular/core/fesm5/core.js.NullInjector.get (core.js:936)
    at resolveToken (core.js:1175)
    at tryResolveToken (core.js:1120)
    at StaticInjector.push../node_modules/@angular/core/fesm5/core.js.StaticInjector.get (core.js:1015)
    at resolveToken (core.js:1175)
    at tryResolveToken (core.js:1120)
    at StaticInjector.push../node_modules/@angular/core/fesm5/core.js.StaticInjector.get (core.js:1015)
    at resolveNgModuleDep (core.js:8065)
    at NgModuleRef_.push../node_modules/@angular/core/fesm5/core.js.NgModuleRef_.get (core.js:8753)
    at resolveDep (core.js:9118)
client?719c:147 [WDS] Warnings while compiling.
warnings @ client?719c:147
onmessage @ socket.js:41
EventTarget.dispatchEvent @ sockjs.js:170
(anonymous) @ sockjs.js:883
SockJS._transportMessage @ sockjs.js:881
EventEmitter.emit @ sockjs.js:86
WebSocketTransport.ws.onmessage @ sockjs.js:2957
wrapFn @ zone.js:1188
push../node_modules/zone.js/dist/zone.js.ZoneDelegate.invokeTask @ zone.js:421
push../node_modules/zone.js/dist/zone.js.Zone.runTask @ zone.js:188
push../node_modules/zone.js/dist/zone.js.ZoneTask.invokeTask @ zone.js:496
invokeTask @ zone.js:1540
globalZoneAwareCallback @ zone.js:1566
client?719c:153 ./node_modules/ng-block-ui/node_modules/@angular/core/esm5/core.js
6570:15-36 Critical dependency: the request of a dependency is an expression
warnings @ client?719c:153
onmessage @ socket.js:41
EventTarget.dispatchEvent @ sockjs.js:170
(anonymous) @ sockjs.js:883
SockJS._transportMessage @ sockjs.js:881
EventEmitter.emit @ sockjs.js:86
WebSocketTransport.ws.onmessage @ sockjs.js:2957
wrapFn @ zone.js:1188
push../node_modules/zone.js/dist/zone.js.ZoneDelegate.invokeTask @ zone.js:421
push../node_modules/zone.js/dist/zone.js.Zone.runTask @ zone.js:188
push../node_modules/zone.js/dist/zone.js.ZoneTask.invokeTask @ zone.js:496
invokeTask @ zone.js:1540
globalZoneAwareCallback @ zone.js:1566
client?719c:153 ./node_modules/ng-block-ui/node_modules/@angular/core/esm5/core.js
6590:15-102 Critical dependency: the request of a dependency is an expression
warnings @ client?719c:153
onmessage @ socket.js:41
EventTarget.dispatchEvent @ sockjs.js:170
(anonymous) @ sockjs.js:883
SockJS._transportMessage @ sockjs.js:881
EventEmitter.emit @ sockjs.js:86
WebSocketTransport.ws.onmessage @ sockjs.js:2957
wrapFn @ zone.js:1188
push../node_modules/zone.js/dist/zone.js.ZoneDelegate.invokeTask @ zone.js:421
push../node_modules/zone.js/dist/zone.js.Zone.runTask @ zone.js:188
push../node_modules/zone.js/dist/zone.js.ZoneTask.invokeTask @ zone.js:496
invokeTask @ zone.js:1540
globalZoneAwareCallback @ zone.js:1566
AppComponent_Host.ngfactory.js? [sm]:1 ERROR TypeError: Cannot read property 'instance' of undefined
    at nodeValue (core.js:8626)
    at Object.eval [as handleEvent] (AppComponent_Host.ngfactory.js? [sm]:1)
    at handleEvent (core.js:9947)
    at callWithDebugContext (core.js:11040)
    at Object.debugHandleEvent [as handleEvent] (core.js:10743)
    at dispatchEvent (core.js:7406)
    at core.js:7850
    at platform-browser.js:1140
    at ZoneDelegate.push../node_modules/zone.js/dist/zone.js.ZoneDelegate.invokeTask (zone.js:421)
    at Object.onInvokeTask (core.js:3645)
View_AppComponent_Host_0 @ AppComponent_Host.ngfactory.js? [sm]:1
proxyClass @ compiler.js:10058
push../node_modules/@angular/core/fesm5/core.js.DebugContext_.logError @ core.js:11002
push../node_modules/@angular/core/fesm5/core.js.ErrorHandler.handleError @ core.js:1567
dispatchEvent @ core.js:7410
(anonymous) @ core.js:7850
(anonymous) @ platform-browser.js:1140
push../node_modules/zone.js/dist/zone.js.ZoneDelegate.invokeTask @ zone.js:421
onInvokeTask @ core.js:3645
push../node_modules/zone.js/dist/zone.js.ZoneDelegate.invokeTask @ zone.js:420
push../node_modules/zone.js/dist/zone.js.Zone.runTask @ zone.js:188
push../node_modules/zone.js/dist/zone.js.ZoneTask.invokeTask @ zone.js:496
invokeTask @ zone.js:1540
globalZoneAwareCallback @ zone.js:1566
postMessage (async)
sendMsg @ client?719c:67
warnings @ client?719c:151
onmessage @ socket.js:41
EventTarget.dispatchEvent @ sockjs.js:170
(anonymous) @ sockjs.js:883
SockJS._transportMessage @ sockjs.js:881
EventEmitter.emit @ sockjs.js:86
WebSocketTransport.ws.onmessage @ sockjs.js:2957
wrapFn @ zone.js:1188
push../node_modules/zone.js/dist/zone.js.ZoneDelegate.invokeTask @ zone.js:421
push../node_modules/zone.js/dist/zone.js.Zone.runTask @ zone.js:188
push../node_modules/zone.js/dist/zone.js.ZoneTask.invokeTask @ zone.js:496
invokeTask @ zone.js:1540
globalZoneAwareCallback @ zone.js:1566
AppComponent_Host.ngfactory.js? [sm]:1 ERROR CONTEXT DebugContext_ {view: {…}, nodeIndex: 0, nodeDef: {…}, elDef: {…}, elView: {…}}

**Comment by kuuurt13** [comment]
@keslavi thanks for the info. So I'm not sure what the issues is but I think it might be specific to your project. I have tested `ng-block-ui` with Angular 6.x using the Angular CLI and it works fine. You can see a sample repo [here](https://github.com/kuuurt13/ng-block-ui-example/tree/ng-6.0). 

I have a few questions/suggestions that might help:
1) Can you try deleting all your `node_modules` and running a fresh `npm install`?
2) Did you upgrade your project with angular cli? As in you were on an older version of angular/cli and recently upgraded by running something like `ng update @angular/cli && ng update @angular/core`? 
3) If you uninstall `ng-block-ui` and/or don't import it into your project does your project still run? It just seems like the first error `ERROR in node_modules/@angular/core/src/render3/ng_dev_mode.d.ts(9,11): error TS2451: Cannot redeclare block-scoped variable 'ngDevMode'.` is a generic error that has nothing to do with `ng-block-ui`

**Comment by kuuurt13** [comment]
After googling your issue I found this as well. It seems like it might be the same issue you are having https://github.com/angular/angular/issues/24165

**Comment by gvart** [comment]
+1 Same issues in Angular 6 project

**Comment by kuuurt13** [comment]
@gvart, glad it is working for you as well. For now`rxjs-compat` is required for Angular 6 apps. Now that Angular 6 has been out for a while, I think I will be releasing `2.0.0` with the correct import paths soon though so people that want to upgrade to 6.0 don't need `rxjs-compat`.

@keslavi, you said you added `rxjs-compat`, correct?

**Comment by tset2411** [comment]
Same issue 

```
Angular CLI: 6.0.8
Node: 8.9.0
OS: linux ia32
Angular: 6.1.0-beta.3
```

`"ng-block-ui": "^1.0.4"`

Error Log:
![screenshot from 2018-07-09 20-03-00](https://user-images.githubusercontent.com/25196573/42457156-4e193afa-83b4-11e8-82ba-ed2fc2148e26.png)

**Comment by kuuurt13** [comment]
@tset2411 thanks for the info. It gives me more info to debug with. I will try to see if I can reproduce.

**Comment by tset2411** [comment]
@kuuurt13 Yeah I have used `ng update` command. Also I have referred this blog [https://www.techiediaries.com/updating-angular-cli-projects/](https://www.techiediaries.com/updating-angular-cli-projects/)

**Comment by tset2411** [comment]
@kuuurt13 I have created new angular 6 project using `ng new Demo` 
Then installed `"ng-block-ui": "^1.0.4"`.
Then I also need to install `npm install rxjs-compat --save` to compile project successfully.
And after I run `ng serve` And It worked in DemoProject

Then I found In my existing project's app module I imported `BlockUIModule` instead of `BlockUIModule.forRoot()`

After changing in app module's import from **`BlockUIModule`** to **`BlockUIModule.forRoot()`** It worked perfectly.
